### PR TITLE
Add Harvester node driver

### PR DESCRIPTION
**Pull request by futuretea**
Add harvester node driver

**Related issues**
harvester/harvester#861
https://github.com/rancher/dashboard/issues/3374

### CODE REVIEW

File: pkg/data/management/machinedriver_data.go [hunk 1]
```diff
@@ -49,7 +51,8 @@ var DriverData = map[string]map[string][]string{
 }
 
 var driverDefaults = map[string]map[string]string{
-	Vmwaredriver: {"vcenterPort": "443"},
+	HarvesterDriver: {"clusterType": "external"},
```
Comment (guangbochen) [inline pkg/data/management/machinedriver_data.go:54]: shouldn't the default should be `imported`?

Comment (guangbochen) [inline pkg/data/management/machinedriver_data.go:54]: shouldn't the default be `imported`?

Comment (futuretea) [reply pkg/data/management/machinedriver_data.go]: fixed

**Review by gitlawr — APPROVED** [review]
LFTM

File: pkg/data/management/machinedriver_data.go [hunk 2]
```diff
@@ -93,6 +96,9 @@ func addMachineDrivers(management *config.ManagementContext) error {
 	if err := addMachineDriver(GoogleDriver, "local://", "", "", nil, false, true, true, management); err != nil {
 		return err
 	}
+	if err := addMachineDriver(HarvesterDriver, "https://harvester-node-driver.s3.amazonaws.com/driver/v0.2.0/docker-machine-driver-harvester-amd64.tar.gz", "", "fdd3428737a15ae28b976476b5b1a4d7d21d9cf9b4c6d9c77cc82e29660f1814", []string{"harvester-node-driver.s3.amazonaws.com"}, false, false, false, management); err != nil {
```
Comment (dramich) [inline pkg/data/management/machinedriver_data.go:99]: Does the harvester UI use this URL `"harvester-node-driver.s3.amazonaws.com"`? If so shouldn't we move this under a rancher/suse owned domain? I saw we were going to embed the UI which is why it's not referenced here.

Comment (guangbochen) [reply pkg/data/management/machinedriver_data.go:99]: yep, the UI is submitted to the rancher dashboard via https://github.com/rancher/dashboard/issues/3374,
file an EIO ticket to move the release to `releases.rancher.com/harvester-node-driver/` and will update this once the release is ready, thanks

**Review by dramich — APPROVED** [review]

**Comment by dramich** [comment]
@guangbochen Can you open a new issue so we can make sure to track changing the whitelisted URL? I merged so we can move forward with testing at least.

**Pull request merged**